solution.js

```diff
@@ -0,0 +1,25 @@
+function lattice(col, row) {
+    if (col === 0 && row === 0) {
+        return 0;
+    }
+    if (col === 0 || row === 0) {
+        return 1;
+    }
+    return lattice(col - 1, row) + lattice(col, row - 1)
+}
+
+function memoize(fn) {
+    const cache = {};
+    return function (...args) {
+        if (cache[args]) {
+            return cache[args];
+        }
+        const result = fn.apply(this, args);
+        cache[args] = result;
+        return result;
+    }
+}
+
+lattice = memoize(lattice);
```

Comment:
One small thing I would suggest here is to not redefining `lattice` value but create another variable instead.
Even I would rather call the function `getLattice() {}` and then create const `lattice`:
```
function getLattice() { ... }
function memoize() { ... }
const lattice = memoize(lattice);
```